ksqldb-engine/src/main/java/io/confluent/ksql/analyzer/Analyzer.java

```diff
@@ -286,12 +286,23 @@ protected AstNode visitJoin(final Join node, final Void context) {
       process(node.getLeft(), context);
       node.getRights().forEach(right -> process(right.getRelation(), context));
 
-      final JoinNode.JoinType joinType = getJoinType(node);
-
       final AliasedDataSource left = analysis.getFromDataSources().get(0);
-      final AliasedDataSource right = analysis.getFromDataSources().get(1);
 
-      final JoinedSource source = Iterables.getOnlyElement(node.getRights());
+      final ImmutableList<JoinedSource> rights = node.getRights();
+      for (int i = 0; i < rights.size(); i++) {
+        final JoinedSource source = rights.get(i);
+        visitJoinedSource(left, analysis.getFromDataSources().get(i + 1), source);
```

Comment:
Just curious, does it matter that you keep passing in left?  What if two of the rights are the same.  Seems like it would miss the self-join check.

Reply:
good catch - I think I need to rework some of this (especially after talking with @vpapavas)